tools/istio-docker.mk

```diff
@@ -255,12 +255,19 @@ docker.save: $(DOCKER_TAR_TARGETS)
 $(foreach TGT,$(DOCKER_TARGETS),$(eval push.$(TGT): | $(TGT) ; \
 	time (docker push $(HUB)/$(subst docker.,,$(TGT)):$(TAG))))
 
+define run_vulnerability_scanning
+        $(eval RESULTS_DIR := vulnerability_scan_results)
+        $(eval CURL_RESPONSE := $(shell curl -s --create-dirs -o $(RESULTS_DIR)/$(1) -w "%{http_code}" http://imagescanner.cloud.ibm.com/scan?image="docker.io/$(2)")) \
+        $(if $(filter $(CURL_RESPONSE), 200), (mv $(RESULTS_DIR)/$(1) $(RESULTS_DIR)/$(1).json), (mv $(RESULTS_DIR)/$(1) $(RESULTS_DIR)/$(1).html))
+endef
+
 # create a DOCKER_PUSH_TARGETS that's each of DOCKER_TARGETS with a push. prefix
 DOCKER_PUSH_TARGETS:=
 $(foreach TGT,$(DOCKER_TARGETS),$(eval DOCKER_PUSH_TARGETS+=push.$(TGT)))
 
 # Will build and push docker images.
 docker.push: $(DOCKER_PUSH_TARGETS)
+	$(foreach TGT,$(DOCKER_TARGETS),$(call run_vulnerability_scanning,$(subst docker.,,$(TGT)),$(HUB)/$(subst docker.,,$(TGT)):$(TAG)))
```

Comment:
If we are going to do this can we please make it opt in (either an env variable or maybe just a separate target). This is used in development and I do not want my images sent to ibm every time I make a change. At the very least it would probably considerably slow down the push

Reply:
completely agree - should be optional.  We have spent alot of time optimizing the build process for speed (and correctness).  I have concerns this would increase build time significantly for a large population of users who build the image when only a few benefit from the CVE scanning.  Can you make this optional default to false?

Reply:
@howardjohn @sdake Thank you for your feedback. It makes sense to me. I will make it optional. I have the sample apps script made optional which I will push in a bit and I will complete the make file and push that after. Thank you once again, appreciate it.